File: Intranet.Web/scripts/Services/VendedorService.js
vendedorApp.service('VendedorService', function ($http) {
    var serviceBase = 'http://localhost:50837/';
    var serviceBaseProducao = 'http://192.168.1.199:9810/Intranet.API/';

    this.GetAll = function () {
        return $http.get(serviceBaseProducao + "api/Vendedor/GetAll").then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.GetAllByNameLike = function (nome) {
        return $http.get(serviceBaseProducao + "api/Vendedor/GetAllByNameLike?nome=" + nome).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.GetAllByNameGrouped = function (nome) {
        return $http.get(serviceBaseProducao + "api/Vendedor/GetAllByNameGrouped?nome=" + nome).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.GetAllByName = function (nome) {
        return $http.get(serviceBaseProducao + "api/Vendedor/GetAllByName?nome=" + nome).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.GetAllFornecedorByName = function (nome) {
        return $http.get(serviceBaseProducao + "api/Vendedor/GetAllFornecedorByName?nome=" + nome).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.GetByEmpresa = function (empresa, nome) {
        return $http.get(serviceBaseProducao + "api/Vendedor/GetByEmpresa?empresa=" + empresa + "&nome=" + nome).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.AddVendedor = function (data) {
        return $http.post(serviceBaseProducao + "api/Vendedor/AddVendedor", data).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.AtualizarVendedor = function (data) {
        return $http.post(serviceBaseProducao + "api/Vendedor/AtulizarVendedor", data).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.IncluirComprador = function (data) {
        return $http.post(serviceBaseProducao + "api/Vendedor/IncluirComprador", data).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.ExcluirComprador = function (data) {
        return $http.post(serviceBaseProducao + "api/Vendedor/ExcluirComprador", data).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.IncluirFornecedor = function (data) {
        return $http.post(serviceBaseProducao + "api/Vendedor/IncluirFornecedor", data).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.ExcluirFornecedor = function (data) {
        return $http.post(serviceBaseProducao + "api/Vendedor/ExcluirFornecedor", data).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }

    this.AlterarStatus = function (data) {
        return $http.post(serviceBaseProducao + "api/Vendedor/AlterarStatus", data).then(function (response) {
            return response
        }, function (response) {
            return alert("Erro: " + response.status);
        });
    }
});

File: Intranet.Service/VendedorService.cs
using Intranet.Domain.Entities;
using Intranet.Domain.Interfaces.Repositories;
using Intranet.Domain.Interfaces.Services;
using Intranet.Domain.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Intranet.Service
{
    public class VendedorService : ServiceBase<Vendedor>, IVendedorService
    {
        private readonly IVendedorRepository _repository;

        public VendedorService(IVendedorRepository repository)
            : base(repository)
        {
            this._repository = repository;
        }

        public IEnumerable<Vendedor> GetVendedorByNome(string nome)
        {
            return _repository.GetVendedorByNome(nome);
        }

        public IEnumerable<Vendedor> GetVendedorByEmpresa(string empresa, string nome)
        {
            return _repository.GetVendedorByEmpresa(empresa, nome);
        }

        public void AtulizarVendedor(Vendedor obj)
        {
            var result = _repository.GetAll().Where(x => x.CdVendedor == obj.CdVendedor);

            foreach (var item in result)
            {
                item.DtAlteracao = DateTime.Now;
                item.Nome = obj.Nome;
                item.DDD = obj.DDD;
                item.Telefone = obj.Telefone;
                item.Email = obj.Email;
                _repository.Update(item);
            }
        }

        public void IncluirComprador(Vendedor obj)
        {
            var result = _repository.Get(x => x.Nome == obj.Nome && x.Empresa == obj.Empresa);

            if (_repository.Get(x => x.Nome == obj.Nome && x.CdComprador == obj.CdComprador) == null)
            {
                result.CdComprador = obj.CdComprador;
                result.DtInclusao = DateTime.Now;
                _repository.Add(result);
            }
        }

        public void ExcluirComprador(Vendedor obj)
        {
            var result = _repository.GetAll().Where(x => x.Nome == obj.Nome && x.CdComprador == obj.CdComprador);

            foreach (var item in result)
            {
                _repository.Remove(item);
            }
        }

        public void IncluirFornecedor(Vendedor obj)
        {
            var result = obj;

            if (_repository.Get(x => x.Nome == obj.Nome && x.Empresa == obj.Empresa) == null)
            {
                result.DtInclusao = DateTime.Now;
                _repository.Add(result);
            }
        }

        public void ExcluirEmpresa(Vendedor obj)
        {
            var result = _repository.GetAll().Where(x => x.Nome == obj.Nome && x.Empresa == obj.Empresa);

            foreach (var item in result)
            {
                _repository.Remove(item);
            }
        }

        public void AlterarStatus(Vendedor obj)
        {
            var result = _repository.GetAll().Where(x => x.CdVendedor == obj.CdVendedor);

            foreach (var item in result)
            {
                item.Status = obj.Status;
                item.DtAlteracao = DateTime.Now;

                _repository.Update(item);
            }
        }
    }
}
